### PR TITLE
Site Transfers: Add margin below Search user and continue button

### PR DESCRIPTION
**Pull request by katinthehatsite**
Closes: https://github.com/Automattic/dotcom-forge/issues/2627

## Proposed Changes

* This PR adds margin below `Search user and continue` button on the site transfer screen under `Settings > General > Transfer your site` to be consistent with [Figma design](1U910ArebC7gbLF1RtxUxE-fi-102_9511)

**Before**:

Screenshot 2023-06-02 at 2 12 26 PM

**After**: 

Screenshot 2023-06-02 at 2 15 51 PM

## Testing Instructions

* Make sure you have a Simple site
* Navigate to `Settings > General > Transfer your site`: https://wpcalypso.wordpress.com/settings/start-site-transfer/{SimpleSite}
* Check that the margin is added below the `Search user and continue` button

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by wojtekn — APPROVED** [review]

File: client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx [hunk 1]
```diff
@@ -22,6 +22,10 @@ const TextControlContainer = styled.div( {
 	},
 } );
 
+const ButtonStyled = styled( Button )`
+	margin-bottom: 1em;
+`;
```
Comment (sejas) [inline client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx]:
```suggestion
const ButtonStyled = styled( Button )( {
	marginBottom: '1em',
} );
```

**Review by sejas — APPROVED** [review]
@katinthehatsite , this change looks good. I tested it on mobile and desktop 👍 .

Can we use JS objects instead of string templates for styling?

File: client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx (hunk 1 above)
Comment (katinthehatsite) [reply client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx]: Sure @sejas ! I am usually seeing both being used so I am newer sure which one I should be using. I remember saying that one of these method is older and one is newer, is that right?

Comment (sejas) [reply client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx]: There is a discussion here p1660141381467549-slack-C0347E545HR.
The TLDR is that with JS objects you get autocomplete, type checker and error linter. The probability of introducing a typo in a CSS attribute is zero.
With string templates we don't get any of those benefits.

**Pull request merged**